sdk/core/azure-core/inc/azure/core/io/body_stream.hpp

```diff
@@ -166,46 +166,58 @@ namespace Azure { namespace IO {
   class FileBodyStream : public BodyStream {
   private:
     // immutable
-#if defined(AZ_PLATFORM_POSIX)
-    int m_fd;
-#elif defined(AZ_PLATFORM_WINDOWS)
-    HANDLE m_hFile;
-#endif
+    FILE* m_fileStream;
     int64_t m_baseOffset;
     int64_t m_length;
+#if defined(AZ_PLATFORM_WINDOWS)
+    int m_fileDescriptor;
+    HANDLE m_filehandle;
+#elif defined(AZ_PLATFORM_POSIX)
+    int m_fileDescriptor;
+#endif
     // mutable
     int64_t m_offset;
 
     int64_t OnRead(Azure::Core::Context const& context, uint8_t* buffer, int64_t count) override;
+    int64_t GetFileSize(FILE* file);
```

Comment:
static?

Reply:
Yes, seems reasonable. I had it static initially but while experimenting with thread safety made it an instance member. Will revert it back to static.

Reply:
I thought we were removing all FILE* from FileBodyStream; it only supports Win32 handles or POSIX fds.